src/mono/netcore/sample/wasm/console/WasmSample.csproj

```
@@ -88,6 +88,16 @@
         <Output TaskParameter="Lines" PropertyName="EmccFlags" />
     </ReadLinesFromFile>
 
+    <ReadLinesFromFile File="$(MicrosoftNetCoreAppRuntimePackDir)native\src\emcc-version.txt">
+        <Output TaskParameter="Lines" PropertyName="RuntimeEmccVersion" />
+    </ReadLinesFromFile>
+
+    <Exec Condition="'$(RunAOTCompilation)' == 'true'" Command="source $(EMSDK_PATH)/emsdk_env.sh &amp;&amp; emcc --version | head -1 > emcc-version.txt" ConsoleToMsBuild="true" IgnoreStandardErrorWarningFormat="true" WorkingDirectory="$(MSBuildThisFileDirectory)$(PublishDir)" />
+    <ReadLinesFromFile Condition="'$(RunAOTCompilation)' == 'true'" File="$(MSBuildThisFileDirectory)$(PublishDir)\emcc-version.txt">
+        <Output TaskParameter="Lines" PropertyName="EmccVersion" />
+    </ReadLinesFromFile>
```

I think this can be simplified to not need a separate file:

```suggestion
    <Exec Condition="'$(RunAOTCompilation)' == 'true'" Command="source $(EMSDK_PATH)/emsdk_env.sh &amp;&amp; emcc --version | head -1" ConsoleToMsBuild="true" IgnoreStandardErrorWarningFormat="true" WorkingDirectory="$(MSBuildThisFileDirectory)$(PublishDir)">
        <Output TaskParameter="ConsoleOutput" PropertyName="EmccVersion" />
    </Exec>
```

---

That won't work because emsdk_env.sh prints out all kinds of output too.